lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -413,6 +414,39 @@ getLocationInfoAsString(const SmallPtrSet<Operation *, 8> &ops) {
   return sstr.str();
 }
 
+/// Emit an optional hardware comment that may exist as an attribute on an
+/// operation.  Return true if a comment was found (and something was written to
+/// the output stream) and false otherwise.
+static bool emitComment(raw_ostream &os, Operation *op) {
+  auto comment =
+      op->getAttrOfType<hw::CommentAttr>(hw::CommentAttr::getMnemonic());
+  if (!comment)
+    return false;
+
+  os << "// " << comment.getText().getValue() << "\n";
```

Comment:
It looks like your design here forces the comment to be a single line, without the `//`.

Given you don't have the `//`, did you consider allowing arbitrary text and wrapping it to 80 columns?  Alternatively, why not include the `//`, allowing the frontend to do full custom formatting however they'd like?

Reply:
This was just the fastest to demo something.  Wrapping to `<num-fill-columns>` is exactly what I want, I just didn't do it in the initial PR. 😅 I would like both support wrapping and promotion to a block comment (if the backend language supports that). There is a design point on if multi-line comment vs. block comment should be a parameter to the `CommentAttr`.

I would like to avoid putting an explicit `//` in the comment as this is intended to be backend agnostic and multi-line vs. block comment for filling purposes. E.g., if the backend is VHDL, then hard-coding `//` would be bad as we VHDL uses `--`.

Reply:
Updated in the rebase to do basic line breaking at before (preferably) or after (as a fallback) the word that goes over the `emittedLineLength`.